src/Schemer/Options.php

```diff
@@ -0,0 +1,571 @@
+	{
+		return $this->addItem($this->items, $item, $key);
+	}
+
+
+	/**
+	 * @param mixed      $item
+	 * @param mixed|null $key
+	 * @return Options
+	 */
+	public function addCandidate($item, $key = null): Options
+	{
+		return $this->addItem($this->candidates, $item, $key);
+	}
+
+
+	/**
+	 * @return array
+	 */
+	public function getItems(): array
+	{
+		return $this->items;
+	}
+
+
+	/**
+	 * @param bool $sortByPriority
+	 * @return Collection
+	 */
+	public function getCandidates($sortByPriority = true): Collection
+	{
+		if (empty($this->candidates)) {
+			return collect();
+		}
+
+		if ($this->containsPrimitives()) {
+			return collect([
+				'key' => collect($this->candidates)
+					->map(function(ArrayItem $item) { return $item->getKey(); })
+					->all()
+			]);
+		}
+
+		/** @var Collection $candidates */
+		$candidates = collect($this->candidates)
+			->map(function(Node $item) { return $item->getChildren(); })
+			->flatten()
+			->filter(function(Property $property) {
+				return !$property->isBag();
+			})
+			->mapWithKeys(function(Property $property) {
+				return [ $property->getName() => $property->getValueProvider() ];
+			});
+
+		if ($sortByPriority === false) {
+			return $candidates;
+		}
+
+		$sortedCandidates = collect();
+
+		$candidates = $candidates
+			->filter(function(ValueProvider $provider = null, $key = null) use (& $sortedCandidates) {
+				if ($provider !== null && $provider->getProperty()->isUniqueKey()) {
+					$sortedCandidates->put($key ,$provider);
+					return false;
+				}
+				return true;
+			})
+			->filter(function(ValueProvider $provider = null, $key = null) use (& $sortedCandidates) {
+				if ($provider !== null && $provider->getProperty()->hasAnyConditionalSiblings()) {
+					$sortedCandidates->put($key ,$provider);
+					return false;
+				}
+				return true;
+			});
+
+		return $sortedCandidates->merge($candidates);
+	}
+
+
+	/**
+	 * @return Property|null
+	 */
+	public function getUniqueKeyProperty(): ?Property
+	{
+		return !$this->containsPrimitives()
+			&& ($candidate = $this->getCandidates()->first()) && $candidate instanceof ValueProvider
+			&& ($property = $candidate->getProperty()) && $property instanceof Property
+			&& $property->isUniqueKey()
+			? $property : null;
+	}
+
+
+	/**
+	 * @param bool $hard
+	 */
+	public function reset(bool $hard = false)
+	{
+		$this->items = [];
+
+		if (empty($this->candidates) || $hard === true) {
+			$this->candidates = [];
+			$this->itemType = $this->associative = null;
+		}
+	}
+
+
+	public function beforeAdded(Node $parent)
+	{
+		$this->validateDeclaration();
+	}
+
+
+	/**
+	 * @param string      $definition
+	 * @param string|null $value
+	 * @param bool        $checkIfExistsOnly
+	 * @return Node|ArrayItem|null
+	 */
+	public function pick($definition, $value = null, $checkIfExistsOnly = false)
+	{
+		if (strpos($definition, '=') !== false) {
+			list($definition, $value) = explode('=', $definition);
```

Comment:
btw dá se to zapsat jako
```suggestion
			[$definition, $value] = explode('=', $definition);
```

Reply:
Done